AirCasting/Graph/GraphView.swift

```diff
@@ -36,35 +36,92 @@ struct GraphView<StatsViewModelType>: View where StatsViewModelType: StatisticsC
-                                                threshold: threshold)
-                    }
+           
+            if procceding(session: session) {
+                    if let threshold = thresholds.threshold(for: selectedStream) {
+                        ZStack(alignment: .topLeading) {
+                            if let selectedStream = selectedStream {
+                                Graph(stream: selectedStream,
+                                      thresholds: threshold,
+                                      isAutozoomEnabled: session.type == .mobile).onDateRangeChange { [weak graphStatsDataSource, weak statsContainerViewModel] range in
+                                        graphStatsDataSource?.dateRange = range
+                                        statsContainerViewModel?.adjustForNewData()
+                                      }
+                                StatisticsContainerView(statsContainerViewModel: statsContainerViewModel,
+                                                        threshold: threshold)
+                            }
 
-                }
-                NavigationLink(destination: HeatmapSettingsView(changedThresholdValues: threshold.rawThresholdsBinding)) {
-                    EditButtonView()
-                }
-                .padding(.horizontal)
-                .padding(.top)
-                
-                ThresholdsSliderView(threshold: threshold)
-                    .padding()
-                    // Fixes labels covered by tabbar
-                    .padding(.bottom)
+                        }
+                        HStack() {
+                            startTimeText
+                            Spacer()
+                            endTimeText
+                        }
+                        .foregroundColor(.aircastingGray)
+                        .padding(.horizontal, 5)
+                        NavigationLink(destination: HeatmapSettingsView(changedThresholdValues: threshold.rawThresholdsBinding)) {
+                            EditButtonView()
+                        }
+                        .padding(.horizontal)
+                        .padding(.top)
+                        
+                        ThresholdsSliderView(threshold: threshold)
+                            .padding()
+                            // Fixes labels covered by tabbar
+                            .padding(.bottom)
+                    }
             }
             Spacer()
         }
         .navigationBarTitleDisplayMode(.inline)
     }
+    
+    var startTimeText: some View {
+        startTime()
+    }
+    
+    var endTimeText: some View {
+        endTime()
+    }
+    
+    func procceding(session: SessionEntity) -> Bool {
+        var showGraph = false
+        guard session.allStreams != nil else { return false }
+        (session.allStreams!.forEach({ stream in
+            (stream.allMeasurements != []) ? (showGraph = true) : (showGraph = false)
```

Comment:
`isEmpty` instead, also can be simplified to just `showGraph = !stream.allMeasurements.isEmpty`. 

But the main point - are you sure it will do the thing you want it to be doing? For me it doesn't make much sense to iterate over all streams when each one will override this `showGraph` value. It's basically equivalent to just doing something like that:
`showGraph = !(session.allStreams.last?.isEmpty ?? true)`

Reply:
Doing this mostly like it is, with some changes from you results in not crashing when entering graph before measurements are loaded so I will change a bit